src/net.cpp

```diff
@@ -1910,8 +1910,8 @@ bool OpenNetworkConnection(const CAddress& addrConnect, bool fCountFailure, CSem
 
 void ThreadMessageHandler()
 {
-    boost::mutex condition_mutex;
-    boost::unique_lock<boost::mutex> lock(condition_mutex);
+    std::mutex condition_mutex;
```

Comment:
So I've actually discovered a nice little trick for this -- 


    struct dummy_mutex {
    void lock(){};
    void unlock(){};
     };

if you want to make it more clear that it's a worthless lock.